java/org/apache/catalina/webresources/AbstractSingleArchiveResourceSet.java

```diff
@@ -60,32 +60,42 @@ public AbstractSingleArchiveResourceSet(WebResourceRoot root, String webAppMount
         }
     }
 
+    private HashMap<String, JarEntry> loadArchiveEntries() throws IOException {
+        JarFile jarFile = null;
+        try {
+            HashMap<String,JarEntry> contents = new HashMap<>();
+            jarFile = openJarFile();
+            System.out.println("MJF ---> AbstractSingleArchiveResourceSet (" + this.getClass().getName() +
+                ") loading archive entries from jar: " + jarFile.getName());
```

Comment:
TODO: remove this diagnostic trace prior to merging

This is showing jars only getting loaded once now, typically at startup when Spring is wiring beans